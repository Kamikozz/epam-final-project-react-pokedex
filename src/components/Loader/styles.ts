import { createStyles } from "@material-ui/core";
import { Theme } from "@material-ui/core/styles";

const styles = (theme: Theme) => createStyles({
  progress: {
    margin: theme.spacing.unit * 2
  },
  progressColor: {
    color: "white"
  }
});

export default styles;
