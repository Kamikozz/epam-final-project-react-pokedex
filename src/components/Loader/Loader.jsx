import React from "react";
import PropTypes from "prop-types";

import { withStyles } from "@material-ui/core/styles";
import CircularProgress from "@material-ui/core/CircularProgress";

const styles = theme => ({
  progress: {
    margin: theme.spacing.unit * 2
  },
  progressColor: {
    color: "white"
  }
});

function Loader(props) {
  const { classes, className, text, size = 70 } = props;
  return (
    <div className={[classes.progressColor, className].join(" ")}>
      {text && <div>LOADING</div>}
      <CircularProgress
        className={classes.progress}
        color="inherit"
        size={size}
        thickness={1}
      />
    </div>
  );
}

Loader.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(Loader);
