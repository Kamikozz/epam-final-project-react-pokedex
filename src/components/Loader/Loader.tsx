import React from "react";
import PropTypes from "prop-types";

import { WithStyles, createStyles } from "@material-ui/core";
import { withStyles, Theme } from "@material-ui/core/styles";
import CircularProgress from "@material-ui/core/CircularProgress";

const styles = (theme: Theme) => createStyles({
  progress: {
    margin: theme.spacing.unit * 2
  },
  progressColor: {
    color: "white"
  }
});

interface Props extends WithStyles<typeof styles> {
  className: string;
  text: string;
  size: number;
  classes: {
    progressColor: string;
    progress: string;
  };
};

function Loader(props: Props) {
  const { classes, className, text, size = 70 } = props;
  return (
    <div className={[classes.progressColor, className].join(" ")}>
      {text && <div>LOADING</div>}
      <CircularProgress
        className={classes.progress}
        color="inherit"
        size={size}
        thickness={1}
      />
    </div>
  );
}

Loader.propTypes = {
  classes: PropTypes.object.isRequired
} as any;

export default withStyles(styles)(Loader);
