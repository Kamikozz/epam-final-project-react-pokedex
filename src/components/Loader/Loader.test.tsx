import React from 'react';
import { render } from '@testing-library/react';

import Loader from './Loader';

it('renders without crashing', () => {
  render(<Loader />);
});

it('matches snapshot', () => {
  const { container } = render(<Loader />);
  expect(container.firstChild).toMatchSnapshot();
});

