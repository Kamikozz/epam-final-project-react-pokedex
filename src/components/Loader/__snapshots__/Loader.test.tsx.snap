// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  class="Loader-progressColor-14 "
>
  <div
    class="MuiCircularProgress-root-15 MuiCircularProgress-indeterminate-17 Loader-progress-13"
    role="progressbar"
    style="width: 70px; height: 70px;"
  >
    <svg
      class="MuiCircularProgress-svg-20"
      viewBox="22 22 44 44"
    >
      <circle
        class="MuiCircularProgress-circle-21 MuiCircularProgress-circleIndeterminate-23"
        cx="44"
        cy="44"
        fill="none"
        r="21.5"
        stroke-width="1"
      />
    </svg>
  </div>
</div>
`;
